xonsh/foreign_shells.py

```diff
@@ -85,27 +85,31 @@
     'zsh': 'env',
     '/bin/zsh': 'env',
     '/usr/bin/zsh': 'env',
+    'cmd': 'set',
 }
 DEFAULT_ALIASCMDS = {
     'bash': 'alias',
     '/bin/bash': 'alias',
     'zsh': 'alias -L',
     '/bin/zsh': 'alias -L',
     '/usr/bin/zsh': 'alias -L',
+    'cmd': 'echo.',
 }
 DEFAULT_FUNCSCMDS = {
     'bash': DEFAULT_BASH_FUNCSCMD,
     '/bin/bash': DEFAULT_BASH_FUNCSCMD,
     'zsh': DEFAULT_ZSH_FUNCSCMD,
     '/bin/zsh': DEFAULT_ZSH_FUNCSCMD,
     '/usr/bin/zsh': DEFAULT_ZSH_FUNCSCMD,
+    'cmd': 'echo.',
```

Comment:
Also, I think think this should be an empty string to indicate that no action is taken.  After some investigation, I couldn't find a way that a batch file could lookup the names of the functions defined in another batch file it has sourced.  This might be impossible, sadly.